src/types.rs

```diff
@@ -24,16 +24,29 @@ const fn hex_val(c: u8) -> Option<u8> {
 
 macro_rules! key {
     ($i:ident, $s:expr) => {
-        #[derive(Default, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+        #[repr(transparent)]
+        #[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
         /// FuelVM atomic type.
         pub struct $i([u8; $s]);
 
+        impl Default for $i {
+            fn default() -> $i {
+                $i([0u8; $s])
```

Comment:
Why remove the proc macro default implementation to add more code that will do the same - but manually instead of relying in the language defaults?